python/tvm/relax/frontend/onnx_frontend.py

```diff
@@ -0,0 +1,1158 @@
+
+    def _check_for_unsupported_ops(self, graph):
+        convert_map = _get_convert_map(self.opset)
+        unsupported_ops = set()
+        for node in graph.node:
+            op_name = node.op_type
+            if (
+                op_name not in convert_map
+                and op_name != "Constant"
+                # and op_name not in _identity_list
+            ):
+                unsupported_ops.add(op_name)
+        if unsupported_ops:
+            msg = "The following operators are not supported for frontend ONNX: "
+            msg += ", ".join(unsupported_ops)
+            raise tvm.error.OpNotImplemented(msg)
+
+    def _construct_nodes(self, graph):
+        """Nodes are stored as directed acyclic graph."""
+        for node in graph.node:
+            op_name = node.op_type
+            attr = self._parse_attr(node.attribute)
+            # Create and populate input list.
+            inputs = onnx_input()
+            for i in node.input:
+                if i != "":
+                    inputs.append(self._nodes[i])
+                else:
+                    inputs.append(None)
+            i_name = self._parse_value_proto(node)
+            outputs = node.output
+            attr["tvm_custom"] = {}
+            attr["tvm_custom"]["name"] = i_name
+            attr["tvm_custom"]["num_outputs"] = len(outputs)
+
+            op = self._convert_operator(op_name, inputs, attr, self.opset)
+
+            if not isinstance(op, relax.Tuple):
+                if isinstance(op.checked_type, tvm.ir.type.TupleType):
+                    # This is a var bound to a tuple. We need to unpack it and create
+                    # a new tuple.
+                    tuple_items = []
+                    for i in range(len(op.checked_type.fields)):
+                        tuple_items.append(self.bb.emit(relax.TupleGetItem(op, i)))
+                    op = relax.Tuple(tuple_items)
+                    outputs_num = len(tuple_items)
+                else:
+                    outputs_num = 1
+            else:
+                outputs_num = len(op)
+            assert (
+                len(outputs) <= outputs_num
+            ), "Missing outputs during conversion. Expected {} but Got {} in {}.".format(
+                len(outputs), outputs_num, op_name
+            )
+
+            if outputs_num == 1:
+                self._nodes[outputs[0]] = op
+            else:
+                for k, i in zip(list(outputs), range(len(outputs))):
+                    self._nodes[k] = op[i]
+
+    def _parse_value_proto(self, value_proto):
+        """Parse ValueProto or raw str."""
+        try:
+            name = value_proto.name
+        except AttributeError:
+            name = value_proto
+        return name
+
+    def _parse_array(self, tensor_proto):
+        np_array = get_numpy(tensor_proto).reshape(tuple(tensor_proto.dims))
+        return tvm.nd.array(np_array)
+
+    def _parse_attr(self, attr_proto):
```

Comment:
This would be a good sort of method to unit test separately, since there are a lot of different cases that we should verify